base/sparse/sparsematrix.jl

```diff
@@ -2893,13 +2862,17 @@ function setindex!{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, x, I::AbstractMatrix{Bool})
                     end
                 end
 
-                # 0: no change, 1: update, 2: delete, 3: add new
-                mode = ((r1 <= r2) && (rowvalA[r1] == row)) ? ((v == 0) ? 2 : 1) : ((v == 0) ? 0 : 3)
+                # 0: no change, 1: update, 3: add new
```

Comment:
Mode 3 renumbered 2 in both relevant methods. Those methods could benefit from a rewrite at some point. Thanks!